proxy/http/HttpTransact.h

```diff
@@ -570,12 +572,60 @@ class HttpTransact
     ConnectionAttributes *server                 = nullptr;
     ink_time_t now                               = 0;
     ServerState_t state                          = STATE_UNDEFINED;
-    unsigned attempts                            = 0;
-    unsigned simple_retry_attempts               = 0;
-    unsigned unavailable_server_retry_attempts   = 0;
-    ParentRetry_t retry_type                     = PARENT_RETRY_NONE;
+    class Attempts
+    {
+    public:
+      // Must only be constructed within a CurrenntInfo instance within a State Instance.
+      Attempts(State const *state_i_am_in);
+      Attempts(Attempts const &) = delete;
+
+      unsigned
+      get() const
+      {
+        return _v;
+      }
+
+      void
+      maximize()
+      {
+        ink_assert(0 == _saved_v);
```

Comment:
I believe this assertion, while generally silenced for release builds, is failing.

Reply:
After more investigation, I see that this assertion is only failing when this patch is applied on top of our HTTP/2 to origin branch.